features/netsocket/Socket.cpp

```diff
@@ -83,11 +94,39 @@ int Socket::modify_multicast_group(const SocketAddress &address, nsapi_socket_op
 
 int Socket::join_multicast_group(const SocketAddress &address)
 {
+    if(_multicast_addr_count == SOCKET_MAX_MULTICAST_MEMBERSHIPS) {
+        debug("Cannot join additional multicast addresses on this socket, increase SOCKET_MAX_MULTICAST_MEMBERSHIPS up to a max of 32");
+        return NSAPI_ERROR_PARAMETER;
+    }
+    uint32_t index = 0;
+    // Find next available multicast address registery slot
+    while((_multicast_addr_registry & (0x0001 << index))) { index++; }
+
+    // Create copy of socketaddress to prevent dead pointers when reading back during leave group
+    _socket_multicast_register[index] = SocketAddress(address);
```

Comment:
It would be a copy-assignment anyway with just `= address`, wouldn't it?

Regardless, I think you would need to be storing the actual multicast request structures, not just the socket address, as you want to track memberships per-interface.

You might want to check for existing matches - add group normally refuses if you've already joined on the same socket. And cancel the entry if the underlying join fails?